trieste/objectives/single_objectives.py

```diff
@@ -383,3 +383,43 @@ def hartmann_6(x: TensorType) -> TensorType:
 
 HARTMANN_6_SEARCH_SPACE = Box([0.0], [1.0]) ** 6
 """ The search space for the :func:`hartmann_6` function. """
+
+
+def michalewicz(x: TensorType, d: int = 2, m: int = 10) -> TensorType:
+    """
+    The Michalewicz function over :math:`[0, \\pi]` for all i=1,...,d. Dimensionality is determined
+    by the parameter ``d`` and it features steep ridges and drops. It has :math:`d!` local minima,
+    and it is multimodal. The parameter ``m`` defines the steepness of they valleys and ridges; a
+    larger ``m`` leads to a more difficult search. The recommended value of ``m`` is 10. See
+    https://www.sfu.ca/~ssurjano/egg.html for details.
+    :param x: The points at which to evaluate the function, with shape [..., d].
+    :return: The function values at ``x``, with shape [..., 1].
+    :raise ValueError (or InvalidArgumentError): If ``x`` has an invalid shape.
+    """
+    tf.debugging.assert_greater_equal(d, 1)
+    tf.debugging.assert_shapes([(x, (..., d))])
+
+    xi = tf.range(1, (d + 1), delta=1, dtype=tf.float64) * tf.pow(x, 2)
+    result = tf.reduce_sum(tf.sin(x) * tf.pow(tf.sin(xi / math.pi), 2 * m), axis=1, keepdims=True)
+
+    return -result
+
+
+MICHALEWICZ_2_MINIMIZER = tf.constant([2.20, 1.57], tf.float64)
+"""
+The global minimizer of the :func:`michalewicz` function over :math:`[0, \\pi]^2`,
+with shape [1, 2] and dtype float64.
+"""
+
+MICHALEWICZ_2_MINIMUM = tf.constant([-1.8013], tf.float64)
+""" The global minimum of the :func:`michalewicz` function, with shape [1] and dtype float64. """
+
+MICHALEWICZ_5_MINIMUM = tf.constant([-4.687658], tf.float64)
+""" The global minimum of the :func:`michalewicz` function, with shape [1] and dtype float64. """
+
+MICHALEWICZ_10_MINIMUM = tf.constant([-9.66015], tf.float64)
+""" The global minimum of the :func:`michalewicz` function, with shape [1] and dtype float64. """
+
```

Comment:
you forgot to define search space variables as well, check the other objective functions